centaur/src/main/scala/centaur/test/formulas/TestFormulas.scala

```diff
@@ -41,18 +41,20 @@ object TestFormulas extends StrictLogging {
   def runSuccessfulWorkflowAndVerifyMetadata(workflowDefinition: Workflow)(implicit cromwellTracker: Option[CromwellTracker]): Test[SubmitResponse] = for {
     _ <- checkDescription(workflowDefinition, validityExpectation = Option(true))
     submittedWorkflow <- runSuccessfulWorkflow(workflowDefinition)
-    metadata <- validateMetadata(submittedWorkflow, workflowDefinition)
-    _ <- validateJobManagerStyleMetadata(submittedWorkflow, metadata.value)
-    _ = cromwellTracker.track(metadata)
-    _ <- validateDirectoryContentsCounts(workflowDefinition, submittedWorkflow, metadata)
+    notArchivedMetadata <- validateMetadata(submittedWorkflow, workflowDefinition, validateArchived = Option(false))
+    _ <- validateJobManagerStyleMetadata(submittedWorkflow, originalMetadata = notArchivedMetadata.value, validateArchived = Option(false))
+    _ = cromwellTracker.track(notArchivedMetadata)
+    _ <- validateDirectoryContentsCounts(workflowDefinition, submittedWorkflow, notArchivedMetadata)
     _ <- waitForArchive(submittedWorkflow, workflowDefinition)
     // Re-validate the metadata now that carboniting has completed
-    _ <- validateMetadata(submittedWorkflow, workflowDefinition)
-    flatMetadata = metadata.asFlat
-    workflowRoot = flatMetadata.value.get("workflowRoot").collectFirst { case JsString(r) => r } getOrElse "No Workflow Root"
+    _ <- validateMetadata(submittedWorkflow, workflowDefinition, validateArchived = Option(true))
```

Comment:
This checks that the fields in the archived metadata still match the expectations in the `.test` file, but not that it's the **exact same** as the response as we got before archival. 

If we can, I wouldn't re-validate against the `.test` file, but instead:
* fetch the new, archived metadata
* use `diffson` to make sure that it's identical to the pre-archival metadata (except for `metadataSource` which will now be `Archived`)

That way, we know that nothing has changed, and, since it passed validation before, it must still be valid wrt the `.test` file.